internal/provider/resource_gitlab_branch.go

```diff
@@ -0,0 +1,236 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_branch", func() *schema.Resource {
+	return &schema.Resource{
+		Description: "This resource allows you to create and manage GitLab branches.",
+
+		CreateContext: resourceGitlabBranchCreate,
+		ReadContext:   resourceGitlabBranchRead,
+		DeleteContext: resourceGitlabBranchDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name for this branch",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"project": {
+				Description: "The ID or full path of the project which the branch is created against",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"ref": {
+				Description: "The ref which the branch is created from",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+			},
+			"web_url": {
+				Description: "The url of the created branch (https)",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"default": {
+				Description: "Bool, true if branch is the default branch for the project",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"can_push": {
+				Description: "Bool, true if you can push to the branch",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"merged": {
+				Description: "Bool, true if the branch has been merged into it's parent",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"commit": {
+				Description: "The commit associated with the branch ref",
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Set:         schema.HashResource(commitSchema),
+				Elem:        commitSchema,
+			},
+		},
+	}
+})
+
+var commitSchema = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"id": {
+			Description: "The unique id assigned to the commit by gitlab.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"author_email": {
+			Description: "The email of the author.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"author_name": {
+			Description: "The name of the author.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"authored_date": {
+			Description: "The date which the commit was authored",
```

Comment:
Let's specify the format in which it is returned.

Reply:
looks resolved